### PR TITLE
feat: support for set cookie

### LINKED ISSUE #1397: add support for set-cookie header
Update `request_context.rs` to hold all `set-cookie` `header` when enabled, and send them in the final response.

```graphql
schema @server(headers: {setCookie: true}) {
  query: Query
}
```

The final response of the API should send combine all the `set-cookie` headers and add it to the final response.

Update the documentation for **server** here: https://tailcall.run/docs/operators/server

### PR DESCRIPTION
**Pull request by ssddOnTop**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #1397
/claim 1397

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Summary by CodeRabbit

- **New Features**
	- Added support for handling cookie headers in server requests, including the ability to set cookies via GraphQL schema and server configuration.
	- Introduced a new field `setCookies` in the GraphQL schema and corresponding logic in server components.
- **Enhancements**
	- Enhanced request context creation with conditional cookie headers addition based on server configuration.
- **Refactor**
	- Refactored logic in the `IO` module to separate setting cache control and cookie headers into distinct functions for improved organization and clarity.

### CODE REVIEW

File: tests/execution/test-set-cookie-headers.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+```yaml @mock
+- request:
+    method: GET
+    url: http://jsonplaceholder.typicode.com/users/1
+
+  response:
+    status: 200
+    body:
```
Comment (tusharmath) [inline tests/execution/test-set-cookie-headers.md:30]: response should have a `set-cookie` header.

Comment (ssddOnTop) [reply tests/execution/test-set-cookie-headers.md:30]: Why? It’s response from jsonplaceholder I dont think we have code that does this

Comment (ssddOnTop) [reply tests/execution/test-set-cookie-headers.md:30]: Was that the original issue?

Comment (tusharmath) [reply tests/execution/test-set-cookie-headers.md:30]: `set-cookie` is a response header and not a request header. It is received from the upstream API responses. The client will never send a `set-cookie` header.

**Review by tusharmath — APPROVED** [review]

**Pull request merged**